src/QsCompiler/Transformations/ClassicallyControlled.cs

```diff
@@ -1002,51 +787,19 @@ private bool IsScopeSingleCall(QsScope contents)
                            && call.Item1.Expression is ExpressionKind.Identifier;
                 }
 
-                public override QsStatementKind OnConjugation(QsConjugation stm)
-                {
-                    var superInWithinBlock = SharedState.InWithinBlock;
-                    SharedState.InWithinBlock = true;
-                    var (_, outer) = this.OnPositionedBlock(QsNullable<TypedExpression>.Null, stm.OuterTransformation);
-                    SharedState.InWithinBlock = superInWithinBlock;
-
-                    var (_, inner) = this.OnPositionedBlock(QsNullable<TypedExpression>.Null, stm.InnerTransformation);
-
-                    return QsStatementKind.NewQsConjugation(new QsConjugation(outer, inner));
-                }
-
-                public override QsStatementKind OnReturnStatement(TypedExpression ex)
-                {
-                    SharedState.IsValidScope = false;
-                    return base.OnReturnStatement(ex);
-                }
-
-                public override QsStatementKind OnValueUpdate(QsValueUpdate stm)
-                {
-                    // If lhs contains an identifier found in the scope's known variables (variables from the super-scope), the scope is not valid
-                    var lhs = this.Expressions.OnTypedExpression(stm.Lhs);
-
-                    if (SharedState.ContainsHoistParamRef)
-                    {
-                        SharedState.IsValidScope = false;
-                    }
-
-                    var rhs = this.Expressions.OnTypedExpression(stm.Rhs);
-                    return QsStatementKind.NewQsValueUpdate(new QsValueUpdate(lhs, rhs));
-                }
-
                 public override QsStatementKind OnConditionalStatement(QsConditionalStatement stm)
                 {
                     var contextValidScope = SharedState.IsValidScope;
-                    var contextHoistParams = SharedState.CurrentHoistParams;
+                    var contextParams = SharedState.GeneratedOpParams;
 
-                    var isHoistValid = true;
+                    var isValidLift = true;
 
                     var newConditionBlocks = new List<Tuple<TypedExpression, QsPositionedBlock>>();
                     var generatedOperations = new List<QsCallable>();
                     foreach (var conditionBlock in stm.ConditionalBlocks)
                     {
                         SharedState.IsValidScope = true;
```

Comment:
It would be nice to have the validation logic all encapsulated in the base LiftContent transformation's LiftBody function. But for now this validation logic is too entangled with the context-specific logic in the LiftConditionBlocks transformation.